Tests/tests.py

```diff
@@ -40,9 +37,29 @@ def assert_matrix(k_expected, k):
     np.testing.assert_allclose(k_expected, k, rtol=1e-3, atol=1e-1)
 
 
-def assert_array1D(g_expected, g):
-    np.testing.assert_allclose(g_expected, g, rtol=1e-3, atol=1e-1)
+def assert_array1D(array1, array2):
+    assert len(array1) == len(array2), "Arrays must be of the same length"
+    for a, b in zip(array1, array2):
+        assert a == b, f"Array values do not match: {a} != {b}"
```

Comment:
I understand this change, but unfortunately, Matlab and python have different accuracy on floats, so we should use assert_allclose.

Reject